pennylane/transforms/split_non_commuting.py

```diff
@@ -271,6 +352,15 @@ def circuit(x):
     if len(tape.measurements) == 0:
         return [tape], null_postprocessing
 
+    if shot_dist is None:
+        shot_dist_fn = None
+    elif isinstance(shot_dist, Callable):
+        shot_dist_fn = shot_dist
+    elif isinstance(shot_dist, str):
+        shot_dist_fn = shot_dist_str2fn.get(shot_dist)
```

Comment:
Let's provide a more informative error here if that key doesn't exist.  Maybe something like
```suggestion
    elif isinstance(shot_dist, str):
        try:
            shot_dist_fn = shot_dist_str2fn.get(shot_dist)
        except KeyError as e:
            raies ValueError(f"Unknown shot_dist {shot_dist}. Available options are {list(shot_dist_str2fn.keys())}.") from e
```
or something